### PR TITLE
[test] Adapt environ regression tests to Eiger

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Review by ekouts — CHANGES_REQUESTED** [review]

File: cscs-checks/prgenv/environ_check.py
Comment (teojgo) [reply cscs-checks/prgenv/environ_check.py:101]: Yeah, that's why I reverted to the original idea after talking to @lucamar

Comment (ekouts) [reply cscs-checks/prgenv/environ_check.py:101]: Ok. If the test is exactly the same, there is no reason to have two different tests, right? You can exclude systems from certain variables like you do for `cray-fftw` and `cray-python`

Comment (ekouts) [reply cscs-checks/prgenv/environ_check.py:101]: Or, if you prefer it, you could make one base test with the logic and then one parameterized for each system (that inherit from the base test).

**Comment by teojgo** [comment]
@ekouts They are more or less the same yes. I have put them in two different classes to be more explicit since the systems are of different nature. Also some of the modules of dom/daint don't exist on eiger and vice-versa.

**Review by vkarak — CHANGES_REQUESTED** [review]
I am refactoring a bit this test.

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:83]: This is ugly. We should wait to merge #1699, which it allows you to override the parameter space of the parent class. You can do it also now with the `parameter()` directive, but the syntax changes in #1699.

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:105]: After #1699 is merged, you will be able to do something like this:

```python
class cray_modules(rfm.RegressionMixin):
    module_name = parameter([...])

class cray_modules_eiger(CrayModules):
    module_name = parameter([...])

class CrayVariablesCheck(rfm.RunOnlyRegressionTest):
     # you base stuff

class CrayVariablesCheckDaint(CrayVariablesCheck, cray_modules):
    ...

class CrayVariablesCheckEiger(CrayVariablesCheck, cray_modules_eiger):
    ...
```

**Comment by teojgo** [comment]
@vkarak I will go on with this now that #1699 is in.

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:61]: I suggest defining the parameter as abstract in the base class, so that nobody can accidentally create a test out of it without defining the parameters.

```suggestion
class CrayVariablesCheck(rfm.RunOnlyRegressionTest):
    cray_modules = parameter()
```

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:76]:
```suggestion
class CrayVariablesCheckDaint(CrayVariablesCheck):
```

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:103]:
```suggestion
class CrayVariablesCheckEiger(CrayVariablesCheck):
```

File: cscs-checks/prgenv/environ_check.py
Comment (vkarak) [inline cscs-checks/prgenv/environ_check.py:119]: Why don't you put the same maintainers in both tests? You may add you and @ekouts.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]
lgtm

**Review by ekouts — APPROVED** [review]

**Pull request merged**
